ocaml/runtime/minor_gc.c

```diff
@@ -411,13 +421,17 @@ static void oldify_mopup (struct oldify_state* st, int do_ephemerons)
     new_v = Field(v, 0);                 /* Follow forward pointer. */
     st->todo_list = Field (new_v, 1);    /* Remove from list. */
 
-    f = Field(new_v, 0);
-    CAMLassert (!Is_debug_tag(f));
-    if (Is_block (f) && Is_young(f)) {
-      oldify_one (st, f, Op_val (new_v));
+    mlsize_t scannable_wosize = Scannable_wosize_val(new_v);
+
+    if (scannable_wosize != 0) {
```

Comment:
`v` comes from `todo_list`, and blocks are now added to `todo_list` only if `scannable_wosize > 1`, so I think this condition is redundant.

Reply:
Indeed. I changed this to a `CAMLassert`. The interplay between `oldify_one` and `oldify_mopup` has confused me once again so I tried to write an explanatory comment on the assert.